packages/services/src/gateway/WorkerManager.ts

```diff
@@ -31,21 +36,37 @@ export default class WorkerManager implements Runner
         return procedureNames.includes(fqn);
     }
 
-    async addWorker(worker: Worker): Promise<void>
+    addWorker(worker: Worker): string
     {
-        this.#workers.add(worker);
+        worker.id = this.#idGenerator.generateUUID();
```

Comment:
```suggestion
        worker.id = this.#idGenerator.generate();
```

It's irrelevant what type Id it gets. That's the responsibility of the generator.